lib/engine/part/city.rb

```diff
@@ -7,15 +7,21 @@ module Engine
   module Part
     class City < RevenueCenter
       attr_accessor :reservations
-      attr_reader :tokens
+      attr_reader :tokens, :bull_tokens
 
       def initialize(revenue, **opts)
         super
         @slots = (opts[:slots] || 1).to_i
         @tokens = Array.new(@slots)
+        # Bull tokens are tokens in a city that don't go in a city slot
+        @bull_tokens = []
         @reservations = []
       end
 
+      def total_slots
```

Comment:
do you need this? could you instead call slots with an arg?

Reply:
I changed it to the way you suggested